app/controllers/site_controller.rb

```diff
@@ -733,17 +786,31 @@ def download_acceptance_params
     params.require(:download_acceptance).permit(:email, :download_agreement_id)
   end
 
+  def validate_reviewer_access_params
+    params.require(:reviewer_access).permit(:pin)
+  end
+
   # make sure user has view permissions for selected study
   def check_view_permissions
     unless @study.public?
-      if (!user_signed_in? && !@study.public?)
+      if !user_signed_in? && @study.reviewer_access.present?
+        reviewer = @study.reviewer_access
+        session_key = cookies.signed[reviewer.cookie_name]
+        if reviewer.expired?
+          alert = 'The review period for this study has expired.'
+          redirect_to merge_default_redirect_params(site_path, scpbr: params[:scpbr]), alert: alert and return
+        elsif session_key.blank? # no cookie present, so this may or may not be a reviewer
+          authenticate_user!
+        elsif !reviewer.session_valid?(session_key) # check session cookie for expiry
+          alert = 'Your review session has expired - please create a new session to continue.'
+          redirect_to merge_default_redirect_params(reviewer_access_path(access_code: reviewer.access_code),
+                                                    scpbr: params[:scpbr]), alert: alert and return
+        end
+      elsif !user_signed_in? && !@study.public?
```

Comment:
I think the study.public? check here is redundant

Reply:
You are correct - we've already checked for it.

Reply:
446268b